src/dispatch/plugins/dispatch_slack/commands.py

```diff
@@ -161,50 +168,68 @@ def filter_tasks_by_assignee_and_creator(tasks: List[Task], by_assignee: str, by
-        if command not in [SLACK_COMMAND_REPORT_INCIDENT_SLUG, SLACK_COMMAND_LIST_INCIDENTS_SLUG]:
-            # We let the user know that incident-specific commands
-            # can only be run in incident conversations
-            return create_command_run_in_nonincident_conversation_message(command)
-
-        # We get the list of public and private conversations the Slack bot is a member of
-        (
-            public_conversations,
-            private_conversations,
-        ) = await dispatch_slack_service.get_conversations_by_user_id_async(
-            client, SLACK_APP_USER_SLUG
+    if (
+        not conversation_name
+        or conversation_name not in public_conversations + private_conversations
+    ):
+        # We let the user know in which public conversations they can run the command
+        return create_command_run_in_conversation_where_bot_not_present_message(
+            command, public_conversations
         )
 
-        # We get the name of conversation where the command was run
-        conversation_id = request.get("channel_id")
-        conversation_name = await dispatch_slack_service.get_conversation_name_by_id_async(
-            client, conversation_id
+    user_id = request.get("user_id")
+    user_email = await dispatch_slack_service.get_user_email_async(client, user_id)
+
+    for f in command_functions(command):
+        background_tasks.add_task(
+            f,
+            user_id,
+            user_email,
+            channel_id,
+            None,
+            organization_slug=organization_slug,
+            command=request,
         )
 
-        if (
-            not conversation_name
-            or conversation_name not in public_conversations + private_conversations
-        ):
-            # We let the user know in which public conversations they can run the command
-            return create_command_run_in_conversation_where_bot_not_present_message(
-                command, public_conversations
-            )
+    return INCIDENT_CONVERSATION_COMMAND_MESSAGE.get(command, f"Running... Command: {command}")
+
+
+async def handle_incident_conversation_commands(client, request, background_tasks):
+    """Handles all commands that are issued from an incident conversation."""
+    channel_id = request.get("channel_id")
+    command = request.get("command")
+    db_session = get_organization_scope_from_channel_id(channel_id=channel_id)
+
+    if not db_session:
+        # We let the user know that incident-specific commands
+        # can only be run in incident conversations
+        return create_command_run_in_nonincident_conversation_message(command)
 
     conversation = conversation_service.get_by_channel_id_ignoring_channel_type(
         db_session=db_session, channel_id=channel_id
     )
 
     user_id = request.get("user_id")
-    user_email = await dispatch_slack_service.get_user_email_async(client, user_id)
+    user_email = dispatch_slack_service.get_user_email_async(client, user_id)
```

Comment:
Have you tested it? AFAIK, `await` would be required here.

Reply:
I did... I'm not sure how it would have worked either. I will add it and double-check.